src/DonorDashboards/resources/js/app/components/subscription-manager/index.tsx

```diff
@@ -37,6 +40,11 @@ const SubscriptionManager = ({id, subscription}) => {
 
     const subscriptionStatus = subscription.payment.status?.id || subscription.payment.status.label.toLowerCase();
 
+    const showAmountControls = wp.hooks.applyFilters(
+        'give_donor_dashboard_subscription_manager_show_amount_controls',
+        true,
+        subscription
+    );
```

Comment:
I think we might be able to simplify this:

What if instead of a front-end filter we tied into the gateway api and added a new front-end gateway property [here](https://github.com/impress-org/give-recurring/blob/c1b04da4dd74cc99bb52a4bfb94bb265930018cd/src/DonorDashboard/Routes/SubscriptionsRoute.php#L146) called 
```php
'can_update_payment_method' => $subscription->can_update()
``` 

Then within this subscription manager we would check for `gateway.can_update` and `gateway.can_update_payment_method` that would show the necessary controls.

Then by simply using the gateway api methods these would show up accordingly. You then wouldn't have to hack that `give_subscription_can_update_subscription` filter in blink as that may actually have some side effects.

Reply:
I completely agree with you. In fact, one thought I had was: What if the `can_update` property were an array specifying what it can update? That way, we would only need to verify what could be affected by the change and ensure it doesn’t break any existing code.

Reply:
Cool, maybe in the future we can go the array route or something similar.  For now, i'd prefer to keep the update properties explicit since we only support updating the amount and payment method at the moment.

Reply:
Implemented on [a36cdee](https://github.com/impress-org/givewp/pull/7632/commits/a36cdee20c1d32b04cba2dc554dc427488f3014d) alongside [e542123](https://github.com/impress-org/give-recurring/pull/1212/commits/e542123a288a8879e8502353eb34f7cec691d50f) on Give Recurring

Reply:
I just realized the gateway properties are coming from recurring, we might want to either use the nullish operator `gateway?.can_update_payment_method` or set some default prop values up the chain in the file when accessing the properties so they don't cause an error if someone is not on the latest version of recurring.

Reply:
Since `gateway` is a prop that we have already implemented, it is not supposed to be nullable. I am only accessing the new `can_update_payment_method` prop, which can be `undefined` at this point, and I am adding a fallback for that case.